vm.c

```diff
@@ -137,19 +128,21 @@ static inline vm_value *vm_get_op_value(vm_env *env, const vm_operand *op)
 void vm_run(vm_env *env)
 {
     size_t pc = 0;
-    BEGIN_OPCODES;
-
-    OP(ADD) : VM_CALL_HANDLER();
-    DISPATCH;
-    OP(SUB) : VM_CALL_HANDLER();
-    DISPATCH;
-    OP(PRINT) : VM_CALL_HANDLER();
-    DISPATCH;
-    OP(JMP) : VM_GOTO(OPCODE.op1.value.id);
-
-    OP(HALT) : goto terminate;
+    for (;;) {
+next_opcode:
```

Comment:
Don't do that. The virtual machine was designed for [threaded interpretation](http://www.ittc.ku.edu/~kulkarni/teaching/EECS768/slides/chapter2.pdf), which implies the elimination of traditional switch-case statement.